CalendarApp/GoogleCalendarService/Manager/EventManager.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Google.Apis.Calendar.v3.Data;
+using GoogleCalendarService.Extensions;
+
+namespace GoogleCalendarService.Manager
+{
+    public class EventManager
+    {        
+        private readonly BookingService _bookingService;
+
+        public EventManager(BookingService bookingService)
+        {
+            _bookingService = bookingService;
+        }
+
+        public List<Event> CheckEvent(string calendarId, Event eventForAdd, bool forcedEvent)
+        {
+            if (forcedEvent)
+                return new List<Event>();
+
+            var eventList = _bookingService.GetEvents(calendarId);
+
+            EventPeriodicityHolder eventForAddHolder = new EventPeriodicityHolder(eventForAdd);
+
+            var allExistedEvents =
+                eventList.SelectMany(ev => new EventPeriodicityHolder(ev).EventList).ToList();
+            var conflictedEvents =
+                eventForAddHolder.EventList.SelectMany(ev => FindConflictedEvents(allExistedEvents, ev)).ToList();
+
+            return conflictedEvents.GroupBy(x => x.Id).Select(x => x.First()).ToList();
```

Comment:
instead of x use abbreviation of the item in the collection u are iterating through. Example:  ev or ce